hazelcast/src/main/java/com/hazelcast/map/impl/MapServiceContextImpl.java

```diff
@@ -181,18 +181,19 @@ public void flushMaps() {
     public void destroyMap(String mapName) {
         final PartitionContainer[] containers = partitionContainers;
         final List<Future> futures = new ArrayList<Future>(containers.length);
+        Address localAddress = getNodeEngine().getLocalMember().getAddress();
         for (PartitionContainer container : containers) {
             if (container != null) {
                 int partitionId = container.getPartitionId();
-                InternalPartition partition = nodeEngine.getPartitionService().getPartition(partitionId);
-
-                if (partition.isLocal()) {
-                    OperationService operationService = nodeEngine.getOperationService();
-                    Operation operation = new MapPartitionDestroyOperation(container, mapName);
-
-                    Future f = operationService.invokeOnPartition(SERVICE_NAME, operation, partitionId);
-                    futures.add(f);
-                }
+                OperationService operationService = nodeEngine.getOperationService();
```

Comment:
Nagging old lady here; I would move the operationService retrievel to the last moment you need it. So first create operation, then retrieve the operationservice and do invoke.